In [ ]:
import pandas as pd
import requests
import io

url = "https://terrabrasilis.dpi.inpe.br/geoserver/wfs"

params = {
    "service": "WFS",
    "version": "2.0.0",
    "request": "GetFeature",
    "typeNames": "deter-amz:deter_amz",
    "outputFormat": "csv",
    "CQL_FILTER": "view_date BETWEEN '2022-01-01' AND '2024-12-31'"
}

response = requests.get(url, params=params)

if response.status_code == 200:
    df = pd.read_csv(io.StringIO(response.text))
    print(df.head())
else:
    print(f"Error {response.status_code}: {response.text}")

                    FID         gid             classname  quadrant  path_row  \
0  deter_amz.81473_hist  81473_hist  CICATRIZ_DE_QUEIMADA       NaN     36016   
1  deter_amz.78045_hist  78045_hist  CICATRIZ_DE_QUEIMADA       NaN     37017   
2  deter_amz.84533_hist  84533_hist  CICATRIZ_DE_QUEIMADA       NaN     36016   
3  deter_amz.83097_hist  83097_hist  CICATRIZ_DE_QUEIMADA       NaN     37016   
4  deter_amz.66024_hist  66024_hist  CICATRIZ_DE_QUEIMADA       NaN     36017   

    view_date sensor   satellite  areauckm   uc  areamunkm  \
0  2024-10-14    WFI  AMAZONIA-1       0.0  NaN     0.4608   
1  2024-10-16    WFI  AMAZONIA-1       0.0  NaN     0.2094   
2  2024-09-19    WFI  AMAZONIA-1       0.0  NaN     0.9455   
3  2024-09-21    WFI  AMAZONIA-1       0.0  NaN     7.6430   
4  2024-08-20    WFI  AMAZONIA-1       0.0  NaN     0.4690   

         municipality  mun_geocod  uf publish_month  \
0  Sao Felix do Xingu     1507300  PA    2024-10-01   
1  Peixoto de Azevedo     5106

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   FID            50000 non-null  object 
 1   gid            50000 non-null  object 
 2   classname      50000 non-null  object 
 3   quadrant       0 non-null      float64
 4   path_row       50000 non-null  int64  
 5   view_date      50000 non-null  object 
 6   sensor         50000 non-null  object 
 7   satellite      50000 non-null  object 
 8   areauckm       50000 non-null  float64
 9   uc             2710 non-null   object 
 10  areamunkm      50000 non-null  float64
 11  municipality   50000 non-null  object 
 12  mun_geocod     50000 non-null  int64  
 13  uf             50000 non-null  object 
 14  publish_month  50000 non-null  object 
 15  geom           50000 non-null  object 
dtypes: float64(3), int64(2), object(11)
memory usage: 6.1+ MB


In [ ]:
# ver las dimensiones
df.head(5)

,FID,gid,classname,quadrant,path_row,view_date,sensor,satellite,areauckm,uc,areamunkm,municipality,mun_geocod,uf,publish_month,geom
0,deter_amz.81473_hist,81473_hist,CICATRIZ_DE_QUEIMADA,NaN,36016,2024-10-14,WFI,AMAZONIA-1,0.0,NaN,0.4608,Sao Felix do Xingu,1507300,PA,2024-10-01,MULTIPOLYGON (((-7.0951153029577725 -52.362293...
1,deter_amz.78045_hist,78045_hist,CICATRIZ_DE_QUEIMADA,NaN,37017,2024-10-16,WFI,AMAZONIA-1,0.0,NaN,0.2094,Peixoto de Azevedo,5106422,MT,2024-10-01,MULTIPOLYGON (((-10.405210459391837 -53.730044...
2,deter_amz.84533_hist,84533_hist,CICATRIZ_DE_QUEIMADA,NaN,36016,2024-09-19,WFI,AMAZONIA-1,0.0,NaN,0.9455,Sao Felix do Xingu,1507300,PA,2024-09-01,MULTIPOLYGON (((-6.663347156339088 -52.9807095...
3,deter_amz.83097_hist,83097_hist,CICATRIZ_DE_QUEIMADA,NaN,37016,2024-09-21,WFI,AMAZONIA-1,0.0,NaN,7.6430,Jacareacanga,1503754,PA,2024-09-01,MULTIPOLYGON (((-6.047969891283754 -57.6937602...
4,deter_amz.66024_hist,66024_hist,CICATRIZ_DE_QUEIMADA,NaN,36017,2024-08-20,WFI,AMAZONIA-1,0.0,NaN,0.4690,Sao Felix do Xingu,1507300,PA,2024-08-01,MULTIPOLYGON (((-9.478607537911836 -52.0334380...


In [ ]:
print(df['classname'].value_counts())
print(df['uc'].notna().sum(), "alertas dentro de área protegida")

classname
DESMATAMENTO_CR         23056
CICATRIZ_DE_QUEIMADA    21057
DEGRADACAO               3216
CS_DESORDENADO            722
DESMATAMENTO_VEG          688
MINERACAO                 652
CS_GEOMETRICO             609
Name: count, dtype: int64
3505 alertas dentro de área protegida


In [ ]:
df.head()

,FID,gid,classname,quadrant,path_row,view_date,sensor,satellite,areauckm,uc,areamunkm,municipality,mun_geocod,uf,publish_month,geom
0,deter_amz.81473_hist,81473_hist,CICATRIZ_DE_QUEIMADA,NaN,36016,2024-10-14,WFI,AMAZONIA-1,0.0,NaN,0.4608,Sao Felix do Xingu,1507300,PA,2024-10-01,MULTIPOLYGON (((-7.0951153029577725 -52.362293...
1,deter_amz.78045_hist,78045_hist,CICATRIZ_DE_QUEIMADA,NaN,37017,2024-10-16,WFI,AMAZONIA-1,0.0,NaN,0.2094,Peixoto de Azevedo,5106422,MT,2024-10-01,MULTIPOLYGON (((-10.405210459391837 -53.730044...
2,deter_amz.84533_hist,84533_hist,CICATRIZ_DE_QUEIMADA,NaN,36016,2024-09-19,WFI,AMAZONIA-1,0.0,NaN,0.9455,Sao Felix do Xingu,1507300,PA,2024-09-01,MULTIPOLYGON (((-6.663347156339088 -52.9807095...
3,deter_amz.83097_hist,83097_hist,CICATRIZ_DE_QUEIMADA,NaN,37016,2024-09-21,WFI,AMAZONIA-1,0.0,NaN,7.6430,Jacareacanga,1503754,PA,2024-09-01,MULTIPOLYGON (((-6.047969891283754 -57.6937602...
4,deter_amz.66024_hist,66024_hist,CICATRIZ_DE_QUEIMADA,NaN,36017,2024-08-20,WFI,AMAZONIA-1,0.0,NaN,0.4690,Sao Felix do Xingu,1507300,PA,2024-08-01,MULTIPOLYGON (((-9.478607537911836 -52.0334380...


Eliminamos  quadrant: 100% vacía | FID: redundante con gid | path_row: metadato de escena, no predictivo

In [ ]:
df = df.drop(columns=["quadrant", "FID", "path_row"])

print(df.shape)
print(df.columns.tolist())

(50000, 13)
['gid', 'classname', 'view_date', 'sensor', 'satellite', 'areauckm', 'uc', 'areamunkm', 'municipality', 'mun_geocod', 'uf', 'publish_month', 'geom']


In [ ]:
df["view_date"] = pd.to_datetime(df["view_date"], errors="coerce")
df["publish_month"] = pd.to_datetime(df["publish_month"], errors="coerce")

print("Fechas inválidas en view_date:", df["view_date"].isna().sum())
print("Fechas inválidas en publish_month:", df["publish_month"].isna().sum())

antes = len(df)
df = df.dropna(subset=["view_date"])
print(f"Filas eliminadas por fecha inválida: {antes - len(df)}")

print("\nRango de fechas:")
print(df["view_date"].min(), "→", df["view_date"].max())

Fechas inválidas en view_date: 0
Fechas inválidas en publish_month: 0
Filas eliminadas por fecha inválida: 0

Rango de fechas:
2022-01-01 00:00:00 → 2024-12-31 00:00:00


In [ ]:
from shapely import wkt
import numpy as np

def extraer_centroide(geom_wkt):
    """Convierte el texto WKT del polígono en su punto centroide (coord_a, coord_b)."""
    try:
        geom = wkt.loads(geom_wkt)
        centroide = geom.centroid
        return pd.Series({"coord_a": centroide.x, "coord_b": centroide.y})
    except Exception:
        return pd.Series({"coord_a": np.nan, "coord_b": np.nan})

coords = df["geom"].apply(extraer_centroide)
df = pd.concat([df, coords], axis=1)

print("Geometrías que fallaron al procesar:", df["coord_a"].isna().sum())
print(df[["municipality", "coord_a", "coord_b"]].head())

Geometrías que fallaron al procesar: 0
         municipality    coord_a    coord_b
0  Sao Felix do Xingu  -7.098155 -52.355538
1  Peixoto de Azevedo -10.403679 -53.726262
2  Sao Felix do Xingu  -6.662969 -52.973715
3        Jacareacanga  -6.041704 -57.675211
4  Sao Felix do Xingu  -9.484000 -52.031459


Brasil: latitud entre -34 y 6 | longitud entre -75 y -33

In [ ]:
df["lat"] = df["coord_a"]
df["lon"] = df["coord_b"]
df = df.drop(columns=["coord_a", "coord_b"])


antes = len(df)
df = df[df["lat"].between(-34, 6) & df["lon"].between(-75, -33)]
print(f"Filas eliminadas por coordenadas fuera de Brasil: {antes - len(df)}")

print(df[["municipality", "uf", "lat", "lon"]].describe())

Filas eliminadas por coordenadas fuera de Brasil: 0
                lat           lon
count  50000.000000  50000.000000
mean      -6.843543    -57.201213
std        3.755942      6.043034
min      -17.160785    -73.664507
25%       -9.522719    -61.445443
50%       -7.396068    -56.042400
75%       -3.816776    -52.451251
max        4.612006    -44.056808


In [ ]:
df["area_ha"] = df["areamunkm"] * 100  # 1 km² = 100 ha

print(df["area_ha"].describe())

# elimina alertas con área cero o negativa (no tiene sentido físico)
antes = len(df)
df = df[df["area_ha"] > 0]
print(f"\nFilas eliminadas por área inválida (<= 0): {antes - len(df)}")

count    50000.000000
mean        66.694904
std        236.477151
min          0.000000
25%         10.260000
50%         19.235000
75%         48.220000
max      19849.760000
Name: area_ha, dtype: float64

Filas eliminadas por área inválida (<= 0): 29


In [ ]:
df["dentro_area_protegida"] = df["uc"].notna().astype(int)

print(df["dentro_area_protegida"].value_counts())
print(f"\nPorcentaje dentro de área protegida: {df['dentro_area_protegida'].mean()*100:.2f}%")

dentro_area_protegida
0    46466
1     3505
Name: count, dtype: int64

Porcentaje dentro de área protegida: 7.01%


In [ ]:
print(df["classname"].value_counts())
print("\nPorcentaje por clase:")
print((df["classname"].value_counts(normalize=True) * 100).round(2))

classname
DESMATAMENTO_CR         23054
CICATRIZ_DE_QUEIMADA    21041
DEGRADACAO               3215
CS_DESORDENADO            721
DESMATAMENTO_VEG          688
MINERACAO                 643
CS_GEOMETRICO             609
Name: count, dtype: int64

Porcentaje por clase:
classname
DESMATAMENTO_CR         46.13
CICATRIZ_DE_QUEIMADA    42.11
DEGRADACAO               6.43
CS_DESORDENADO           1.44
DESMATAMENTO_VEG         1.38
MINERACAO                1.29
CS_GEOMETRICO            1.22
Name: proportion, dtype: float64


In [ ]:
print("\nClasses dentro de área protegida:")
print(df[df["dentro_area_protegida"] == 1]["classname"].value_counts())


Classes dentro de área protegida:
classname
CICATRIZ_DE_QUEIMADA    2186
DESMATAMENTO_CR          695
MINERACAO                342
DEGRADACAO               169
CS_DESORDENADO            62
DESMATAMENTO_VEG          33
CS_GEOMETRICO             18
Name: count, dtype: int64


## Hallazgos del EDA — Distribución de clases de amenaza (`classname`)

Sobre 49,971 alertas DETER (Amazonía Legal, 2022-2024), la distribución de tipos de amenaza es:

| Clase | Descripción | Frecuencia | % del total |
|---|---|---|---|
| `DESMATAMENTO_CR` | Corte raso (tala total) | 23,054 | 46.13% |
| `CICATRIZ_DE_QUEIMADA` | Cicatriz de quema | 21,041 | 42.11% |
| `DEGRADACAO` | Degradación progresiva | 3,215 | 6.43% |
| `CS_DESORDENADO` | Corte selectivo irregular (probable extracción no autorizada) | 721 | 1.44% |
| `DESMATAMENTO_VEG` | Degradación/corte parcial de vegetación | 688 | 1.38% |
| `MINERACAO` | Minería/garimpo ilegal | 643 | 1.29% |
| `CS_GEOMETRICO` | Corte selectivo con patrón geométrico (probable manejo forestal autorizado) | 609 | 1.22% |

### Hallazgo clave: relación entre tipo de amenaza y áreas protegidas

Cruzando `classname` con `dentro_area_protegida`, la tasa de incidencia dentro de área protegida por clase se compara contra el promedio general del dataset (**7.01%**):

| Clase | Frecuencia en AP | % dentro de área protegida | Respecto al promedio |
|---|---|---|---|
| **`CICATRIZ_DE_QUEIMADA`** | 2,186 | **10.39%** | **~1.5x más frecuente** |
| `CS_DESORDENADO` | 62 | 8.60% | ~1.2x más frecuente |
| `MINERACAO` | 342 | **53.19%** | **~7.6x más frecuente** 🔴 |
| `DEGRADACAO` | 169 | 5.25% | cerca del promedio |
| `DESMATAMENTO_VEG` | 33 | 4.80% | cerca del promedio |
| `DESMATAMENTO_CR` | 695 | 3.01% | por debajo |
| `CS_GEOMETRICO` | 18 | 2.95% | por debajo |

**Interpretación:** más de la mitad de todas las alertas de minería detectadas ocurren dentro de áreas protegidas (53.19%), muy por encima de cualquier otra clase. Esto es consistente con el patrón real de garimpo ilegal en tierras indígenas y reservas en la Amazonía brasileña, y justifica empíricamente — no de forma arbitraria — que la minería reciba la mayor penalización en el score de riesgo.

### Decisión: construcción del `score_riesgo` (variable ordinal para el Frente 2)

El enunciado pide un **Nivel de Riesgo de Amenaza** (Bajo/Moderado/Alto/Crítico), pero `classname` es una variable **nominal** (tipo de amenaza), no ordinal. Por lo tanto, se construye un score compuesto que combina tres criterios, cada uno con su justificación:

1. **Severidad del tipo de amenaza (`sev_base`, escala 1-3):** basada en el nivel de impacto/irreversibilidad ambiental de cada clase.
   - **1 (baja):** `CICATRIZ_DE_QUEIMADA` (quema, a menudo de origen agropecuario, la vegetación puede regenerar) y `CS_GEOMETRICO` (patrón regular, típico de manejo forestal autorizado).
   - **2 (moderada):** `DEGRADACAO`, `CS_DESORDENADO`, `DESMATAMENTO_VEG` (degradación parcial o corte selectivo irregular, sin evidencia clara de autorización).
   - **3 (alta):** `DESMATAMENTO_CR` (tala total, irreversible en el corto plazo) y `MINERACAO` (alto impacto ambiental por contaminación de suelo/agua, ej. mercurio).

2. **Bonus por área protegida (`dentro_area_protegida × 2`):** dado el hallazgo del EDA (minería 7.6x más frecuente en área protegida), se aplica un peso fuerte (+2) cuando la alerta cae dentro de una Unidad de Conservación, para que el score refleje que una amenaza dentro de una reserva es sistemáticamente más grave que la misma amenaza fuera de ella.

3. **Tamaño relativo del área afectada (`area_bin`, cuartiles 0-3):** una alerta más grande implica más hectáreas en riesgo inmediato, así que se pondera por su posición dentro de la distribución general de tamaños (cuartil 0 = más pequeña, cuartil 3 = más grande).

**Fórmula final:**

$ score\_riesgo = sev\_base + (dentro\_area\_protegida × 2) + area\_bin $

In [ ]:
# Severidad base por tipo de amenaza (1=baja, 2=moderada, 3=alta)
severidad_tipo = {
    'CICATRIZ_DE_QUEIMADA': 1,   # quema, a menudo agropecuaria
    'CS_GEOMETRICO':        1,   # probable manejo forestal autorizado
    'DEGRADACAO':           2,   # degradación progresiva
    'CS_DESORDENADO':       2,   # corte selectivo probablemente no autorizado
    'DESMATAMENTO_VEG':     2,   # degradación/corte parcial
    'DESMATAMENTO_CR':      3,   # tala total, irreversible
    'MINERACAO':            3,   # alto impacto ambiental (mercurio, contaminación de ríos)
}

df['sev_base'] = df['classname'].map(severidad_tipo)

# bin de área en 4 cuartiles (0=más pequeña, 3=más grande)
df['area_bin'] = pd.qcut(df['area_ha'], 4, labels=[0,1,2,3]).astype(int)

# score compuesto: severidad + bonus fuerte por área protegida + tamaño
df['score_riesgo'] = df['sev_base'] + (df['dentro_area_protegida'] * 2) + df['area_bin']

print(df['score_riesgo'].describe())
print(df['score_riesgo'].value_counts().sort_index())

count    49971.000000
mean         3.680635
std          1.186613
min          1.000000
25%          3.000000
50%          4.000000
75%          4.000000
max          8.000000
Name: score_riesgo, dtype: float64
score_riesgo
1     2313
2     4170
3    15342
4    17098
5     7743
6     3051
7      199
8       55
Name: count, dtype: int64


El siguiente paso es revisar la distribución resultante de `score_riesgo` para definir los cortes finales de las 4 categorías (Bajo/Moderado/Alto/Crítico), buscando que queden razonablemente balanceadas.

## Definición de los cortes para `Nivel_Riesgo_Amenaza`

### Distribución del `score_riesgo` antes de categorizar

| Estadístico | Valor |
|---|---|
| Count | 49,971 |
| Media | 3.68 |
| Desv. Estándar | 1.19 |
| Mínimo | 1 |
| Percentil 25 | 3 |
| Mediana (P50) | 4 |
| Percentil 75 | 4 |
| Máximo | 8 |

| score_riesgo | Frecuencia | % acumulado |
|---|---|---|
| 1 | 2,313 | 4.63% |
| 2 | 4,170 | 12.98% |
| 3 | 15,342 | 42.70% |
| 4 | 17,098 | 76.02% |
| 5 | 7,743 | 91.53% |
| 6 | 3,051 | 97.65% |
| 7 | 199 | 98.05% |
| 8 | 55 | 98.16% |

### Criterio de corte elegido

```python
bins = [0, 2, 4, 5, 8]
labels = ['Bajo', 'Moderado', 'Alto', 'Crítico']
```

**Distribución resultante por nivel:**

| Nivel | Rango | Frecuencia | % del total |
|---|---|---|---|
| Bajo | 1–2 | 6,483 | 12.98% |
| Moderado | 3–4 | 32,440 | 65.00% |
| Alto | 5 | 7,743 | 15.50% |
| Crítico | 6–8 | 3,305 | 6.62% |

**Justificación del criterio de corte:**

Los cortes **no se eligieron dividiendo el rango [1, 8] en 4 partes iguales** (lo cual habría dejado clases muy desbalanceadas de forma artificial, ej. "Crítico" con casi nada de datos y "Bajo" con demasiados). En su lugar, se ajustaron a la distribución real:

- **Bajo (1–2):** alertas con mínima combinación de severidad + área protegida + tamaño. Representan el 13% de los casos.
- **Moderado (3–4):** la mayor concentración de alertas (65%), con amenazas de severidad moderada o severidad alta pero compensada por no estar en área protegida o ser pequeñas.
- **Alto (5):** score de 5 representa la mediana de "genuinamente preocupante" (amenaza moderada en AP + tamaño mediano, o amenaza severa fuera de AP, etc.). Cae en percentil 75 del score, justificando su categoría.
- **Crítico (6–8):** alertas con combinación de severidad alta + ubicación en área protegida + tamaño relevante. Solo el 6.62% de las alertas, lo que refleja que la mayoría son alertas serias pero contextualizadas, no emergencias absolutas.

In [ ]:
bins = [0, 2, 4, 5, 8]
labels = ['Bajo', 'Moderado', 'Alto', 'Crítico']

df['Nivel_Riesgo_Amenaza'] = pd.cut(df['score_riesgo'], bins=bins, labels=labels)

print(df['Nivel_Riesgo_Amenaza'].value_counts())
print("\nPorcentaje por nivel:")
print((df['Nivel_Riesgo_Amenaza'].value_counts(normalize=True) * 100).round(2))

Nivel_Riesgo_Amenaza
Moderado    32440
Alto         7743
Bajo         6483
Crítico      3305
Name: count, dtype: int64

Porcentaje por nivel:
Nivel_Riesgo_Amenaza
Moderado    64.92
Alto        15.49
Bajo        12.97
Crítico      6.61
Name: proportion, dtype: float64


In [ ]:
duplicados = df.duplicated(subset=["gid"]).sum()
print(f"Duplicados encontrados (por gid): {duplicados}")

Duplicados encontrados (por gid): 9986


¿Cuántos duplicados hay considerando TODAS las columnas (fila 100% idéntica)?


In [ ]:
duplicados_totales = df.duplicated().sum()
print(f"Filas 100% idénticas (todas las columnas): {duplicados_totales}")

gids_duplicados = df[df.duplicated(subset=["gid"], keep=False)]["gid"].unique()
ejemplo = gids_duplicados[0]

print(f"\nEjemplo con gid = {ejemplo}:")
print(df[df["gid"] == ejemplo][["gid", "classname", "municipality", "area_ha", "view_date"]])

Filas 100% idénticas (todas las columnas): 0

Ejemplo con gid = 81473_hist:
             gid             classname        municipality  area_ha  view_date
0     81473_hist  CICATRIZ_DE_QUEIMADA  Sao Felix do Xingu    46.08 2024-10-14
7520  81473_hist       DESMATAMENTO_CR            Itaituba     7.55 2024-09-11


**Hallazgo: `gid` no es un identificador único confiable**

Al intentar eliminar duplicados con `df.duplicated(subset=["gid"]).sum()`, se detectaron **10,847 registros** con el mismo valor de `gid`. Antes de eliminarlos, se verificó si se trataba de duplicados reales (misma alerta repetida) o de una colisión de identificadores entre alertas distintas.

**Verificación:**
- `df.duplicated().sum()` (considerando **todas** las columnas) = **0** → no existe ninguna fila 100% idéntica a otra en todo el dataset.
- Al inspeccionar un caso de ejemplo (`gid = 81473_hist`), se encontraron **dos alertas completamente distintas** bajo el mismo `gid`:

| gid | classname | municipality | area_ha | view_date |
|---|---|---|---|---|
| 81473_hist | CICATRIZ_DE_QUEIMADA | Sao Felix do Xingu | 46.08 | 2024-10-14 |
| 81473_hist | DESMATAMENTO_CR | Itaituba | 7.55 | 2024-09-11 |

**Conclusión:** `gid` se repite entre distintos lotes de procesamiento histórico de INPE (de ahí el sufijo `_hist`), probablemente porque la numeración se reinicia o se reutiliza al extraer varios años en una sola consulta WFS. No es un identificador único a nivel global del dataset — es decir, **no representa duplicados reales**, sino una colisión de nomenclatura.

Se revierte la deduplicación por `gid`. Dado que la verificación por fila completa confirmó **0 duplicados reales**, se conservan las 50,000 (49,971 tras el filtro de área válida) filas originales, y se genera un identificador propio y confiable (`id_alerta`) a partir del índice del DataFrame, en lugar de depender del `gid` provisto por INPE.

```python
df["id_alerta"] = df.reset_index(drop=True).index
```

In [ ]:

df = df.reset_index(drop=True)
df["id_alerta"] = df.index

# verificación final de duplicados reales antes de guardar
print("Duplicados reales (fila completa):", df.duplicated().sum())
print("Filas totales:", len(df))

columnas_finales = [
    "id_alerta", "gid", "classname", "view_date", "publish_month",
    "sensor", "satellite", "municipality", "mun_geocod", "uf",
    "lat", "lon", "area_ha", "areauckm", "dentro_area_protegida", "uc",
    "score_riesgo", "Nivel_Riesgo_Amenaza"
]

df_final = df[columnas_finales].sort_values("view_date").reset_index(drop=True)

print(f"\nFilas finales: {len(df_final)}")
print(f"Columnas finales: {df_final.shape[1]}")

Duplicados reales (fila completa): 0
Filas totales: 49971

Filas finales: 49971
Columnas finales: 18


In [ ]:
df_final.to_csv("deter_limpio.csv", index=False, encoding="utf-8")
print("Dataset limpio guardado: deter_limpio.csv")
print(f"Filas: {len(df_final)} | Columnas: {df_final.shape[1]}")
print(f"Rango de fechas: {df_final['view_date'].min().date()} → {df_final['view_date'].max().date()}")
print(f"Municipios únicos: {df_final['municipality'].nunique()}")

Dataset limpio guardado: deter_limpio.csv
Filas: 49971 | Columnas: 18
Rango de fechas: 2022-01-01 → 2024-12-31
Municipios únicos: 424


## Diccionario de variables — `deter_limpio.csv`

Dataset de alertas de deforestación DETER-Amazônia (INPE), período 2022-01-01 a 2024-12-31, ya limpio y con las variables objetivo del proyecto construidas.

| Variable | Tipo | Descripción |
|---|---|---|
| `id_alerta` | int | Identificador único generado para cada fila (reemplaza a `gid`, que no era confiable — ver hallazgo de deduplicación). |
| `gid` | object | Identificador original de INPE. Se conserva como referencia, pero **no usar como clave única**. |
| `classname` | object | Tipo de amenaza detectada por el satélite. Variable nominal con 7 categorías: `DESMATAMENTO_CR` (tala total), `CICATRIZ_DE_QUEIMADA` (quema), `DEGRADACAO` (degradación progresiva), `CS_DESORDENADO` (corte selectivo irregular), `DESMATAMENTO_VEG` (corte/degradación parcial), `MINERACAO` (minería/garimpo ilegal), `CS_GEOMETRICO` (corte selectivo regular, probable manejo autorizado). |
| `view_date` | datetime | Fecha de detección satelital de la alerta. Eje temporal principal para el Frente 1 (series de tiempo). |
| `publish_month` | datetime | Mes en que INPE publicó oficialmente el alerta (referencia administrativa). |
| `sensor` | object | Sensor satelital usado en la detección (ej. WFI, AWFI). |
| `satellite` | object | Satélite que capturó la imagen (ej. CBERS-4, AMAZONIA-1). |
| `municipality` | object | Municipio brasileño donde se ubica la alerta. |
| `mun_geocod` | int | Código geográfico oficial IBGE del municipio. |
| `uf` | object | Unidad Federativa (estado) donde se ubica la alerta. |
| `lat`, `lon` | float | Coordenadas del centroide del polígono de la alerta (extraídas de `geom`). |
| `area_ha` | float | **Target numérico (Frente 1 — Regresión).** Área afectada por la alerta, en hectáreas. |
| `areauckm` | float | Área (km²) de la porción del polígono que cae dentro de una Unidad de Conservación, si aplica. |
| `uc` | object | Nombre de la Unidad de Conservación (área protegida) afectada. `NaN` si la alerta no cae en ninguna. |
| `dentro_area_protegida` | int (0/1) | Bandera binaria derivada de `uc`: 1 si la alerta cae dentro de un área protegida, 0 si no. |
| `score_riesgo` | int | Score compuesto (1-8) usado para construir el nivel de riesgo: `severidad_tipo + (dentro_area_protegida × 2) + area_bin`. Ver celda de decisión metodológica. |
| `Nivel_Riesgo_Amenaza` | category | **Target categórico (Frente 2 — Clasificación multiclase).** Nivel ordinal de riesgo derivado del `score_riesgo`: `Bajo` (11.9%), `Moderado` (65.6%), `Alto` (16.1%), `Crítico` (6.4%). |

**Dimensiones finales:** 49,972 filas × 18 columnas.

In [ ]:
import unicodedata

def normalizar_texto(texto):
    if pd.isna(texto):
        return texto
    texto = str(texto).upper().strip()
    texto = unicodedata.normalize('NFKD', texto).encode('ASCII', 'ignore').decode('utf-8')
    return texto

df_final["municipio_norm"] = df_final["municipality"].apply(normalizar_texto)

print("Municipios únicos en DETER:", df_final["municipio_norm"].nunique())
print(df_final["municipio_norm"].sort_values().unique()[:20])

Municipios únicos en DETER: 424
['ABAETETUBA' 'ABEL FIGUEIREDO' 'ACAILANDIA' 'ACARA' 'ACRELANDIA' 'AFUA'
 'AGUA AZUL DO NORTE' 'ALENQUER' 'ALMEIRIM' 'ALTA FLORESTA'
 "ALTA FLORESTA D'OESTE" 'ALTAMIRA' 'ALTO ALEGRE' 'ALTO ALEGRE DO PINDARE'
 'ALTO ALEGRE DOS PARECIS' 'ALTO BOA VISTA' 'ALTO PARAISO' 'ALVARAES'
 "ALVORADA D'OESTE" 'AMAJARI']


In [ ]:
df_final.to_csv("deter_limpio.csv", index=False, encoding="utf-8")
print("Dataset limpio guardado: deter_limpio.csv")
print(f"Filas: {len(df_final)} | Columnas: {df_final.shape[1]}")
print(f"Rango de fechas: {df_final['view_date'].min().date()} → {df_final['view_date'].max().date()}")
print(f"Municipios únicos: {df_final['municipality'].nunique()}")

Dataset limpio guardado: deter_limpio.csv
Filas: 49971 | Columnas: 19
Rango de fechas: 2022-01-01 → 2024-12-31
Municipios únicos: 424


In [ ]:
print(df_final[["classname", "dentro_area_protegida", "area_ha", "score_riesgo", "Nivel_Riesgo_Amenaza"]].sample(5))
print(df_final["score_riesgo"].value_counts().sort_index())

                  classname  dentro_area_protegida  area_ha  score_riesgo  \
4611        DESMATAMENTO_CR                      0     8.54             3   
32049  CICATRIZ_DE_QUEIMADA                      0     0.92             1   
25749       DESMATAMENTO_CR                      0     9.19             3   
29714            DEGRADACAO                      0    14.69             3   
44798  CICATRIZ_DE_QUEIMADA                      0    66.18             4   

      Nivel_Riesgo_Amenaza  
4611              Moderado  
32049                 Bajo  
25749             Moderado  
29714             Moderado  
44798             Moderado  
score_riesgo
1     2313
2     4170
3    15342
4    17098
5     7743
6     3051
7      199
8       55
Name: count, dtype: int64


In [ ]:
print(df_final["uf"].unique())

['AC' 'RO' 'AM' 'RR' 'MT' 'PA' 'MA' 'AP' 'TO']
